### DISCUSSION TITLE
SMA Tripower X inverters

### DISCUSSION THREAD

**Discussion by Bimboboy007** [SBFspot/SBFspot#634] [Q&A]
Are the SMA Tripower X inverters also supported?

**Comment by sbf-** [comment]
All SMA inverters talking SMAdata2+ over Bluetooth or speedwire should work

**Comment by Bimboboy007** [comment]
These too ?
Sunny Tripower X 12 -- STP 12-50

Sunny Tripower X 15 -- STP 15-50

**Reply by sbf-** [reply]
Not sure as these are brand new types.
If you have them already installed I'd say give it a try.

**Comment by derron7** [comment]
Connection to X 15 model fails with:
```
Sun Apr 16 14:26:17 2023: INFO: Starting...
sunrise: 06:15
sunset : 20:11
Connecting to Local Network...
Initialising...
SUSyID: 125 - SessionID: 843042988
Device IP address: 192.168.16.33 from config
--------: 00 01 02 03 04 05 06 07 08 09
00000000: 53 4D 41 00 00 04 02 A0 00 00 
00000010: 00 01 00 26 00 10 60 65 09 A0 
00000020: FF FF FF FF FF FF 00 00 7D 00 
00000030: AC D0 3F 32 00 00 00 00 00 00 
00000040: 01 80 00 02 00 00 00 00 00 00 
00000050: 00 00 00 00 00 00 00 00 
58 Bytes sent to [192.168.16.33]
Timeout reading socket
No data!
ERROR: Connection to inverter failed!
Is 192.168.16.33 a correct IP?
Sun Apr 16 14:26:19 2023: CRITICAL: Failed to initialise Speedwire connection.
Sun Apr 16 14:26:19 2023: INFO: Done.
```
Ping works. ~~Sunny Explorer connection works but without any data.~~ (edit: Connection also works with wrong password. May be Sunny Explorer is too old for the X models) Inverter only installation without SHM.
As far as I know Speedwire is active (default) and I am not able to change it? I set Modbus to active in the web interface.

**Reply by sbf-** [reply]
That's what I was afraid of...
Maybe try with `IP_Address=0.0.0.0`


> Connection also works with wrong password. May be Sunny Explorer is too old for the X models

That's normal, even for older model inverters: wrong password: connection OK, but no data (greyed out inverter icon in SE)

Maybe you have to take the modbus route... plenty of projects on github

**Reply by derron7** [reply]
With 0.0.0.0 the inverter was found, but with same result:
```
Connecting to Local Network...
Initialising...
SUSyID: 125 - SessionID: 938689976
Valid response from SMA device 192.168.16.33
ERROR: Connection to inverter failed!
Is 192.168.16.33 a correct IP?
Sun Apr 16 15:43:43 2023: CRITICAL: Failed to initialise Speedwire connection.
Sun Apr 16 15:43:43 2023: INFO: Done.
```
I also configured Modbus. It works with limitations, currently only a few values are accessable. With my older STP-15000 I used the Speedwire logging a long time without any problems.

**Reply by derron7** [reply]
I just heard from the SMA Hotline (1st Level) that the Tripower X models do not communicate with the Sunny Explorer tool. I was also told that Speedwire is only used to search for other devices. But I don't quite believe the latter. I have now opened a ticket to hopefully reach second level support.

**Comment by aoverweel** [comment]
I just installed the Sunny STP 15-50 but not tested yet with SBFspot. It seems that new inverters are running on EnnexOS. The onboarding to SMA is totally different, so assuming that SBFspot does not work also. The inverters are also not to be onboarded to sunny explorer but to the new portal ennexos.sunnyportal.com.

**Comment by sbf-** [comment]
According to [this document](https://files.sma.de/downloads/STPxx-50-SystemManager-TI-en-10.pdf) on page 8 "Communication/Protocols" SMAData2+ is supported. In this case SunnyExplorer and SBFspot should be able to talk to the STP X inverters.

![image](https://github.com/SBFspot/SBFspot/assets/1931158/aa4b2e1a-56dd-4bfb-9245-6a346f163333)

I wonder if this communication protocol is (like Modbus) disabled by default and has to be activated in order to make it work with SE/SBFspot

**Reply by whistlefromthesun** [reply]
SMA Data2+ is always enabled it is the native language of SMA devices.

**Reply by sbf-** [reply]
That's what I think also... but why SE is not able to communicate with this type of devices? Encrypted data?

**Comment by whistlefromthesun** [comment]
![Ohne Titel](https://github.com/SBFspot/SBFspot/assets/152595103/1ff5c037-b63c-4417-b9a6-a9167826bfac)

Just some hints...

[Recordtypes.zip](https://github.com/SBFspot/SBFspot/files/13645999/Recordtypes.zip)

**Reply by sbf-** [reply]
> Recordtypes.zip

Official SMA documentation or...?

I was able to decompile Sunnyexplorer revealing the same info:
```
using System;
using System.Text;
namespace SMA.Data.Odb.Types.Generated
{
	public class AboPullEntry : OdbRecord
	{
		public U32 JobNr;
		public U16 ObjNum;
		public U08 TimeRequest;
		public U08 Spare;
		public U32 IdxFirst;
		public U32 IdxLast;
		public AboPullEntry()
		{
			this.JobNr = new U32();
			this.ObjNum = new U16();
			this.TimeRequest = new U08();
			this.Spare = new U08();
			this.IdxFirst = new U32();
			this.IdxLast = new U32();
		}
		public override int Size()
		{
			return 16;
		}
		public override int Serialize(byte[] buffer, ref int pos)
		{
<snip>
```

**Reply by raimund-hdr** [reply]
somebody has posted more info on SMA protocol (maybe those information can help):
https://github.com/wimleers/homebridge-sma-home-manager/issues/16

**Comment by bacardischmal** [comment]
is there hope that something has changed in the firmware version from 03.02.08.R to 3.02.11.R to help on this issue?

**Reply by SBFspot** [reply]
No, firmware update will not change anything…
If you can make it work with sunnyexplorer, it can work with SBFspot

@whistlefromthesun SBFspot (and SE) are using device-cast (SUSy id=0xFFFF) Why are these X inverters not responding? It seems you have a lot of useful SMAdata2+ information, but you remain so vague…
